core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1167,7 +1168,14 @@ class KafkaApis(val requestChannel: RequestChannel,
     var topicMetadata =
       if (authorizedTopics.isEmpty)
         Seq.empty[MetadataResponse.TopicMetadata]
-      else {
+      else if (metadataRequest.excludePartitions()) {
+        val topicsOnlyMetadata = new ArrayBuffer[MetadataResponse.TopicMetadata]
+        val emptyPartitionMetadata = new util.ArrayList[MetadataResponse.PartitionMetadata]()
```

Comment:
nit - this can be made into a static const

Reply:
Thanks, it's been changed to a static final field inside the MetadataResponse class.